routing/http/client/client.go

```diff
@@ -96,10 +116,13 @@ func New(baseURL string, opts ...option) (*client, error) {
 	return client, nil
 }
 
-func (c *client) FindProviders(ctx context.Context, key cid.Cid) (provs []types.ProviderResponse, err error) {
+func (c *client) FindProviders(ctx context.Context, key cid.Cid) (provs iter.Iter[types.ProviderResponse], err error) {
 	measurement := newMeasurement("FindProviders")
 	defer func() {
-		measurement.length = len(provs)
+		if sliceIter, ok := provs.(*iter.SliceIter[types.ProviderResponse]); ok {
+			length := len(sliceIter.Slice)
+			measurement.length = &length
+		}
```

Comment:
We could send a measurement to the iterator and add a counter to measure received elements on close or on finish.